### ISSUE TITLE
Too many resolutions for Live transcoding triggers rate limiting from remote runnner

### ISSUE THREAD

**Issue by agates** [Chocobozzz/PeerTube#5834]
### Describe the current behavior

Enabling many (I tested with 5-7) live stream resolutions in `peertube-runner` appears to be running into API rate limiting, resulting in a 429 response

```
[02:12:58.323] ERROR (8): Cannot process job
    err: {
      "type": "Error",
      "message": "Expected status 204, got 429. \nThe server responded: \"Too many requests, please try again later.\".\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
      "stack":
          Error: Expected status 204, got 429. 
          The server responded: "Too many requests, please try again later.".
          You may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs
              at buildRequest (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84939:14)
              at makeUploadRequest (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84865:9)
              at RunnerJobsCommand.postUploadRequest (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:85023:12)
              at RunnerJobsCommand.update (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:89791:17)
              at ProcessLiveRTMPHLSTranscoding.<anonymous> (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91322:46)
              at Generator.next (<anonymous>)
              at /usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:69:61
              at new Promise (<anonymous>)
          ----
              at /usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84942:19
              at /usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53819:17
              at Test._assertFunction (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53805:17)
              at Test.assert (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53695:27)
              at localAssert (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53663:18)
              at /usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53667:11
              at Request2.callback (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53236:7)
              at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53388:22)
              at IncomingMessage.emit (node:events:525:35)
              at endReadableNT (node:internal/streams/readable:1359:12)
      "res": {
        "req": {
          "method": "POST",
          "url": "https://redacted/api/v1/runners/jobs/0a9dba20-7f36-4566-bf4f-dc7ae17bc9fe/update",
          "headers": {}
        },
        "header": {
          "server": "nginx/1.25.0",
          "date": "Wed, 14 Jun 2023 02:12:58 GMT",
          "content-type": "text/html; charset=utf-8",
          "content-length": "42",
          "connection": "close",
          "x-powered-by": "PeerTube",
          "x-frame-options": "DENY",
          "tk": "N",
          "access-control-allow-origin": "*",
          "access-control-allow-credentials": "true",
          "access-control-expose-headers": "Retry-After",
          "x-ratelimit-limit": "50",
          "x-ratelimit-remaining": "0",
          "x-ratelimit-reset": "1686708779",
          "retry-after": "10",
          "etag": "W/\"2a-UpTsLJ74nYuiLgNgEwlQMxGqwrE\""
        },
        "status": 429,
        "text": "Too many requests, please try again later."
      }
    }
```

### Steps to reproduce

1. Enable remote runners
2. Register at least one remote runner
3. Enable all transcoding resolutions
4. Test with a 1080p or higher stream to ensure enough resolutions are generated
5. After a few minutes, notice the PeerTube runner jobs error out.  The stream should restart.

### Describe the expected behavior

The expected behavior is to not trigger rate limiting

### Additional information

* PeerTube instance:
  * Version: 5.2.0-rc.1
  *peertube-runner NodeJS version:  18.16.0
  *peertube-runner Ffmpeg version: 5.1.3-1

* Browser name, version and platforms on which you could reproduce the bug:  Installed peertube-runner in a container running on kubernetes https://hub.docker.com/r/alecksgates/peertube-runner ([dockerfile repo](https://github.com/agates/peertube-runner-container/tree/main))

**Comment by agates** [comment]
Here are some logs where one occurrence of rate limiting appears to start (timestamp 2023-06-19 17:41:04.751).  I don't see anything obvious, but if it helps, the runner and stream all came from the same source IP along with open web browser tabs watching several streams/videos.

Tried to reproduce locally with no success, but I did triple check that the given rate limiting code is running on the server.  Will keep trying.

[peertube-runner-rate-limit.log](https://github.com/Chocobozzz/PeerTube/files/11796383/peertub-runner-rate-limit.log)
